test_runner/fixtures/utils.py

```diff
@@ -86,7 +86,15 @@ def __init__(self, in_file, out_file, echo: bool, capture: bool):
             self.captured = ""
 
         def run(self):
+            first = True
             for line in self.in_file:
+                if first:
+                    # do this only after receiving any input so that we can keep deleting empty files
+                    first = False
+                    # prefix the files with the command line so that we can later understand which file is for what command
+                    cmdline = " ".join(cmd)
+                    self.out_file.write(("# " + cmdline + "\n").encode("utf-8"))
```

Comment:
We can use f-strings here

```suggestion
                    cmdline = " ".join(cmd)
                    self.out_file.write(f"# #{cmdline}\n".encode("utf-8"))
```

Or even 
```suggestion
                    self.out_file.write(f"# #{' '.join(cmd)}\n".encode("utf-8"))
```

Reply:
Aah yes it will still require the quotes to be different, no biggie.